### ISSUE TITLE
createPortableExecutable error

### ISSUE THREAD

**Issue by lemuellew** [NationalSecurityAgency/ghidra#3800]
script
```
            ByteProvider byteProvider = new MemoryByteProvider(currentProgram.getMemory(), currentProgram.getImageBase());
            PortableExecutable pe = PortableExecutable.createPortableExecutable(RethrowContinuesFactory.INSTANCE,
                    byteProvider, PortableExecutable.SectionLayout.MEMORY);
            header = pe.getNTHeader().getOptionalHeader();
```

exception stack
```
java.io.IOException: Unable to read bytes at ram:180020c00
        at ghidra.app.util.bin.MemoryByteProvider.readBytes(MemoryByteProvider.java:145)
        at ghidra.app.util.bin.BinaryReader.readLong(BinaryReader.java:699)
        at ghidra.app.util.bin.format.pe.ThunkData.initThunkData(ThunkData.java:83)
        at ghidra.app.util.bin.format.pe.ThunkData.createThunkData(ThunkData.java:69)
        at ghidra.app.util.bin.format.pe.ImportAddressTableDataDirectory.parse(ImportAddressTableDataDirectory.java:99)
        at ghidra.app.util.bin.format.pe.DataDirectory.processDataDirectory(DataDirectory.java:80)
        at ghidra.app.util.bin.format.pe.ImportAddressTableDataDirectory.initImportAddressTableDataDirectory(ImportAddressTableDataDirectory.java:52)
        at ghidra.app.util.bin.format.pe.ImportAddressTableDataDirectory.createImportAddressTableDataDirectory(ImportAddressTableDataDirectory.java:42)
        at ghidra.app.util.bin.format.pe.OptionalHeaderImpl.processDataDirectories(OptionalHeaderImpl.java:460)
        at ghidra.app.util.bin.format.pe.NTHeader.parse(NTHeader.java:279)
        at ghidra.app.util.bin.format.pe.NTHeader.initNTHeader(NTHeader.java:94)
        at ghidra.app.util.bin.format.pe.NTHeader.createNTHeader(NTHeader.java:75)
        at ghidra.app.util.bin.format.pe.PortableExecutable.initPortableExecutable(PortableExecutable.java:110)
        at ghidra.app.util.bin.format.pe.PortableExecutable.createPortableExecutable(PortableExecutable.java:86)
        at ghidra.app.util.bin.format.pe.PortableExecutable.createPortableExecutable(PortableExecutable.java:69)

```

**Comment by ryanmkurtz** [comment]
Probably a similar problem to #3347, since `SectionLayout.MEMORY` is being used.

**Comment by ryanmkurtz** [comment]
I've confirmed that it is fixed by #3347.
